x-pack/plugins/snapshot_restore/public/app/services/navigation/breadcrumb.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { BASE_PATH } from '../../constants';
+import { textService } from '../text';
+
+class BreadcrumbService {
+  private chrome: any;
+  private breadcrumbs: any = {
+    management: {},
+    home: {},
+    repositoryAdd: {},
+    repositoryEdit: {},
+  };
+
+  public init(chrome: any, managementBreadcrumb: any): void {
+    this.chrome = chrome;
+    this.breadcrumbs.management = managementBreadcrumb;
+    this.breadcrumbs.home = {
+      text: textService.breadcrumbs.home,
+      href: `#${BASE_PATH}`,
+    };
+    this.breadcrumbs.repositoryAdd = {
+      text: textService.breadcrumbs.repositoryAdd,
+      href: `#${BASE_PATH}/add_repository`,
```

Comment:
FWIW, the last breadcrumb in the breadcrumb trail doesn't need an `href`, since there's no reason for the user to click it. This does get  bit tricker with the `home` breadcrumb though. We might need to add some logic to conditionally apply the `href` if it's not the last one.